### PR TITLE
Add useProxies to registryMirror allowing to mirror more anything

### PR DESCRIPTION
**Pull request by momiji**
## Description

Add a new field in registryMirror, allowing to use proxies list to decide whether to use dragonfly.
By doing this, it is now possible to cache more than only blobs/sha256 urls, allowing to use dragonfly2 for other usage thatn docker registries, like yum repositories.

Also fix the whiteList test which is tested on direct request, although host is empty and canno tbe tested.

Please update the documentation in Chinese, I just can't do it.

## Related Issue

#959 

## Motivation and Context

This PR now allows to use dragonfly to mirror yum repositories, at least.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist:

- [x] Code compiles correctly.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: client/daemon/proxy/proxy.go [hunk 1]
```diff
@@ -556,7 +559,14 @@ func (proxy *Proxy) shouldUseDragonfly(req *http.Request) bool {
...
 func (proxy *Proxy) shouldUseDragonflyForMirror(req *http.Request) bool {
-	return proxy.registry != nil && !proxy.registry.Direct && transport.NeedUseDragonfly(req)
+	if proxy.registry == nil || proxy.registry.Direct {
+		return false
+	}
+	if proxy.registry.UseProxies {
+		return proxy.shouldUseDragonfly(req)
+	} else {
+		return transport.NeedUseDragonfly(req)
+	}
```
Comment (jim3ma) [inline client/daemon/proxy/proxy.go:569]:
```
	if proxy.registry.UseProxies {
		return proxy.shouldUseDragonfly(req)
	}
	return transport.NeedUseDragonfly(req)
```

Comment (momiji) [reply client/daemon/proxy/proxy.go:567]: It's done, I just hope I didn't screwed up anything with the signoff thing...

**Comment by momiji** [comment]
I apologize as I've never done this before, but I guess once used to the PR process, it will be easier and less a pain for me :)

I've tested the fix and confirm is it working perfectly on my real environments, using the docker image built for this purpose on my github account.

Will there be a v2.0.2 or v2.1.0 version including the fix soon? Not sure if including the alpha changes is necessary.

**Comment by jim3ma** [comment]
> I apologize as I've never done this before, but I guess once used to the PR process, it will be easier and less a pain for me :)
> 
> I've tested the fix and confirm is it working perfectly on my real environments, using the docker image built for this purpose on my github account.
> 
> Will there be a v2.0.2 or v2.1.0 version including the fix soon? Not sure if including the alpha changes is necessary.

Next release is v2.0.2. This change will be included.

**Review by jim3ma — APPROVED** [review]
LGTM

**Review by gaius-qi — APPROVED** [review]
LGTM
